core/amber/src/main/scala/edu/uci/ics/texera/workflow/common/workflow/WorkflowCompiler.scala

```diff
@@ -73,10 +76,12 @@ class WorkflowCompiler(
     //  by cache.
     val originalLogicalPlan = compileLogicalPlan(logicalPlanPojo, executionStateStore)
 
+    assert(originalLogicalPlan.isDefined, "logical plan compilation failed")
+
     // the cache-rewritten LogicalPlan. It is considered to be equivalent with the original plan.
     val rewrittenLogicalPlan = WorkflowCacheRewriter.transform(
       context,
-      originalLogicalPlan,
+      originalLogicalPlan.get,
```

Comment:
you could get an unwanted error from getting a None.

Reply:
reverted the change.